.github/workflows/DevEnv-backend-BuildImageAndPush.yml

```diff
@@ -1,36 +1,90 @@
+
+      # Build from docker file. Tags the build with two tags for use in the next step which pushes them to the registry. One tag for tools and one for dev. Only 'tags' are used as this allows specifying
+      # the image name in each tag. This is needed because it is different in each namespace as it contains '-dev' suffix in the dev namespace. Note, with multiple tags both have the same digest value.
+      - name: Build from Dockerfile
+        id: build-docker
+        uses: redhat-actions/buildah-build@v2
+        with:
+          tags: ${{ env.TOOLSNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.IMAGE_TAGS }} ${{ env.DEVNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}${{ env.DEV_SUFFIX }}:${{ env.IMAGE_TAGS }}
+          context: ./backend
+          dockerfiles: |
+            ./backend/Dockerfile
+
+      # Push both images at the same time using different tags from the build. This allows having different image names.
+      - name: Push to tools and dev registry
+        id: push-images
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          tags: ${{ steps.build-docker.outputs.tags }}
+          username: ${{ env.IMAGE_REGISTRY_USER }}
+          password: ${{ env.IMAGE_REGISTRY_PASSWORD }}
+
+      # Update the image ID in the manifest repository used by Argo CD.
+      # We need to use the SSH deploy key to check out the manifest repo, set up
+      # Kustomize in the runner, update the image ID, commit the change, and
+      # push the changes back to the manifest repository.
+      # ------------------------------------------------------------------------
+      - name: Prepare deploy key
+        uses: webfactory/ssh-agent@v0.5.3
+        with:
+          ssh-private-key: ${{ secrets.GITOPS_C5839F_PRIVATE_KEY_TEMP }}
+
+      - name: Check out manifest repo
+        uses: actions/checkout@v2
+        with:
+          ssh-key: ${{ secrets.GITOPS_C5839F_PRIVATE_KEY_TEMP }}
+          repository: ${{ secrets.GITOPS_C5839F_REPO }}
+          ref: 6016-Setup-user-transfer
 
-      # Runs a single command using the runners shell
-      - name: Run a one-line script
-        run: echo Hello, world!
+      - name: Set up Kustomize
+        uses: imranismail/setup-kustomize@v1
 
-      # Runs a set of commands using the runners shell
-      - name: Run a multi-line script
+      - name: Update image ID and commit change
+        if: ${{ success() }}
         run: |
-          echo Add other actions to build,
-          echo test, and deploy your project.
+          cd account-transfer-devops/overlays/dev
+          kustomize edit set image "backend-image=image-registry.openshift-image-registry.svc:5000/c5839f-dev/${{ env.IMAGE_NAME }}${{ env.DEV_SUFFIX }}@${{ steps.push-images.outputs.digest }}"
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git commit -am "Update Backend API image ID for Dev"
```

Comment:
We should be able to add deployment version here in comments as well

Reply:
added version to comment